File: hooks/before-agent.js
#!/usr/bin/env node
/**
 * BeforeAgent Hook: Display pending tasks
 * Windows 11 Compatible
 */

const fs = require('fs');
const path = require('path');
const os = require('os');

async function main() {
  try {
    const input = JSON.parse(fs.readFileSync(0, 'utf-8'));
    
    // Get task list ID
    const taskListId = process.env.GEMINI_TASK_LIST_ID || 'default';
    
    // Windows-compatible path
    const tasksDir = path.join(os.homedir(), '.gemini', 'tasks');
    const tasksFile = path.join(tasksDir, `${taskListId}.json`);
    
    if (!fs.existsSync(tasksFile)) {
      console.log(JSON.stringify({}));
      return;
    }
    
    const tasks = JSON.parse(fs.readFileSync(tasksFile, 'utf-8'));
    const pending = tasks.filter(t => t.status === 'pending' || t.status === 'in_progress');
    
    if (pending.length === 0) {
      console.log(JSON.stringify({}));
      return;
    }
    
    // Build task display
    let taskDisplay = '\n📋 Active Tasks:\n';
    pending.slice(0, 10).forEach(task => {
      const icon = task.status === 'in_progress' ? '⟳' : '○';
      const blockers = task.blocked_by?.length > 0 
        ? ` [blocked by: ${task.blocked_by.join(', ')}]` 
        : '';
      taskDisplay += `  ${icon} ${task.id}: ${task.subject}${blockers}\n`;
    });
    
    if (pending.length > 10) {
      taskDisplay += `  ... and ${pending.length - 10} more\n`;
    }
    
    // Inject as additional context
    console.log(JSON.stringify({
      systemMessage: taskDisplay,
      hookSpecificOutput: {
        hookEventName: "BeforeAgent",
        additionalContext: taskDisplay
      }
    }));
    
  } catch (error) {
    console.error('BeforeAgent hook error:', error.message);
    console.log(JSON.stringify({}));
  }
}

main();

File: hooks/before_agent.py
#!/usr/bin/env python3
import json
import os
import sys
from pathlib import Path

def main():
    try:
        # Read input from stdin
        try:
            line = sys.stdin.read()
            if not line:
                print(json.dumps({}))
                return
            input_data = json.loads(line)
        except json.JSONDecodeError:
            print(json.dumps({}))
            return

        # Get task list ID
        task_list_id = os.environ.get("GEMINI_TASK_LIST_ID", "default")
        
        # Windows-compatible path
        tasks_dir = Path.home() / ".gemini" / "tasks"
        tasks_file = tasks_dir / f"{task_list_id}.json"
        
        if not tasks_file.exists():
            print(json.dumps({}))
            return
        
        with open(tasks_file, "r", encoding="utf-8") as f:
            tasks = json.load(f)
            
        pending = [t for t in tasks if t.get("status") in ["pending", "in_progress"]]
        
        if not pending:
            print(json.dumps({}))
            return
            
        # Build task display with proper newlines
        task_display = "\n📋 Active Tasks:\n"
        for task in pending[:10]:
            icon = "⟳" if task.get("status") == "in_progress" else "○"
            blockers_list = task.get("blocked_by", [])
            blockers = f" [blocked by: {', '.join(blockers_list)}]" if blockers_list else ""
            task_display += f"  {icon} {task.get('id')}: {task.get('subject')}{blockers}\n"
            
        if len(pending) > 10:
            task_display += f"  ... and {len(pending) - 10} more\n"
            
        # Inject as additional context
        print(json.dumps({
            "systemMessage": task_display,
            "hookSpecificOutput": {
                "hookEventName": "BeforeAgent",
                "additionalContext": task_display
            }
        }))
        
    except Exception as e:
        sys.stderr.write(f"BeforeAgent hook error: {str(e)}\n")
        print(json.dumps({}))

if __name__ == "__main__":
    main()